test/e2e/util.go

```diff
@@ -242,7 +243,7 @@ func waitForPodRunningInNamespace(c *client.Client, podName string, namespace st
 			return true, nil
 		}
 		if pod.Status.Phase == api.PodFailed {
-			return true, fmt.Errorf("Giving up; pod went into failed status: \n%#v", pod.Status)
+			return true, fmt.Errorf("Giving up; pod went into failed status: \n%s", spew.Sprintf("%#v", pod))
```

Comment:
I'm assuming it was in a non-failed state and flipped to failed, and more than one field of the pod status changed reflecting why it went to failed. Direct pod spew is sometimes hard to parse, especially with multiple containers.